js/server/modules/@arangodb/foxx/manager.js

```diff
@@ -848,22 +830,35 @@ function install (serviceInfo, mount, options = {}) {
       `
     });
   }
-  const tempPaths = _prepareService(serviceInfo, options);
-  _install(mount, tempPaths, options);
+  const tempPaths = _prepareService(serviceInfo, options.legacy);
+  const tempService = FoxxService.create({
+    mount,
+    basePath: tempPaths.tempServicePath,
+    noisy: true,
+    options: {
+      development: options.development,
+      configuration: options.configuration,
+      dependencies: options.dependencies
+    }
+  });
+  _install(tempService, tempPaths.tempBundlePath, {
+    setup: options.setup !== false,
+    force: options.force === true
+  });
   selfHeal();
   propagateSelfHeal();
+  reloadRouting();
 
-  return FoxxService.create({
-    mount,
-    options
-  });
+  return getServiceInstance(mount);
 }
 
 function uninstall (mount, options = {}) {
   ensureFoxxInitialized();
-  const service = _uninstall(mount, options);
+  const oldService = _uninstall(mount, options);
+  selfHeal();
   propagateSelfHeal();
-  return service;
+  reloadRouting();
```

Comment:
Considering we're doing the `selfHeal();propagateSelfHeal();reloadRouting();` dance in each of these functions now, we might want to extract those into a function at some point.